libfwupd/fwupd-common.c

```diff
@@ -537,39 +536,187 @@ typedef struct __attribute__((packed)) {
  * Since: 1.2.5
  **/
 gchar *
-fwupd_guid_from_buf (const guint8 *buf, FwupdGuidFlags flags)
+fwupd_guid_to_string (const fwupd_guid_t *guid, FwupdGuidFlags flags)
 {
-	FwupdGuid guid;
+	fwupd_guid_native_t gnat;
+
+	g_return_val_if_fail (guid != NULL, NULL);
 
 	/* copy to avoid issues with aligning */
-	memcpy (&guid, buf, sizeof(FwupdGuid));
+	memcpy (&gnat, guid, sizeof(gnat));
 
 	/* mixed is bizaar, but specified as the DCE encoding */
 	if (flags & FWUPD_GUID_FLAG_MIXED_ENDIAN) {
 		return g_strdup_printf ("%08x-%04x-%04x-%04x-%02x%02x%02x%02x%02x%02x",
-					GUINT32_FROM_LE(guid.a),
-					GUINT16_FROM_LE(guid.b),
-					GUINT16_FROM_LE(guid.c),
-					GUINT16_FROM_BE(guid.d),
-					guid.e[0], guid.e[1],
-					guid.e[2], guid.e[3],
-					guid.e[4], guid.e[5]);
+					GUINT32_FROM_LE(gnat.a),
+					GUINT16_FROM_LE(gnat.b),
+					GUINT16_FROM_LE(gnat.c),
+					GUINT16_FROM_BE(gnat.d),
+					gnat.e[0], gnat.e[1],
+					gnat.e[2], gnat.e[3],
+					gnat.e[4], gnat.e[5]);
 	}
 	return g_strdup_printf ("%08x-%04x-%04x-%04x-%02x%02x%02x%02x%02x%02x",
-				GUINT32_FROM_BE(guid.a),
-				GUINT16_FROM_BE(guid.b),
-				GUINT16_FROM_BE(guid.c),
-				GUINT16_FROM_BE(guid.d),
-				guid.e[0], guid.e[1],
-				guid.e[2], guid.e[3],
-				guid.e[4], guid.e[5]);
+				GUINT32_FROM_BE(gnat.a),
+				GUINT16_FROM_BE(gnat.b),
+				GUINT16_FROM_BE(gnat.c),
+				GUINT16_FROM_BE(gnat.d),
+				gnat.e[0], gnat.e[1],
+				gnat.e[2], gnat.e[3],
+				gnat.e[4], gnat.e[5]);
+}
+
+#if !GLIB_CHECK_VERSION(2,54,0)
+static gboolean
+str_has_sign (const gchar *str)
+{
+	return str[0] == '-' || str[0] == '+';
+}
+
+static gboolean
+str_has_hex_prefix (const gchar *str)
+{
+	return str[0] == '0' && g_ascii_tolower (str[1]) == 'x';
 }
 
+static gboolean
+g_ascii_string_to_unsigned (const gchar *str,
+			    guint base,
+			    guint64 min,
+			    guint64 max,
+			    guint64 *out_num,
+			    GError **error)
+{
+	const gchar *end_ptr = NULL;
+	gint saved_errno = 0;
+	guint64 number;
+
+	g_return_val_if_fail (str != NULL, FALSE);
+	g_return_val_if_fail (base >= 2 && base <= 36, FALSE);
+	g_return_val_if_fail (min <= max, FALSE);
+	g_return_val_if_fail (error == NULL || *error == NULL, FALSE);
+
```

Comment:
I'd think you also want a check for `out_num` here no?

Reply:
I just copied it from GLib :) I'm tempted to just fix it upstream for things like this. In an ideal world I'd bump the GLib dep, but that might make me unpopular...

Reply:
Yeah, I think holding off as long as possible on bumping that particular dep is ideal.